# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/MacroAssembler.h
# Commit: a5cd8e8ce4b5
# Full Hash: a5cd8e8ce4b5be97e0e8aab8483719bb3e93b08f
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-11-03 21:39:33
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 2: Preserve callee and caller tls'es. r=lth
#   
#   This is a followup patch for removing Frame::tls.
#   Now, we are preserving caller's and callee's tls'es for all possible cross-instance calls in the previously allocated abi slots.
#   We also use preserved tls values to restore the caller's tls in Ion. Baseline doesn't need this because it restores the caller tls from its private stack slot after the call.
# ==============================================================================

diff -r 8e95ddbb6b20 -r a5cd8e8ce4b5 js/src/jit/MacroAssembler.h
--- a/js/src/jit/MacroAssembler.h	Tue Nov 03 10:18:03 2020 +0000
+++ b/js/src/jit/MacroAssembler.h	Tue Nov 03 10:20:49 2020 +0000
@@ -260,6 +260,11 @@
 
 enum class CharEncoding { Latin1, TwoByte };
 
+constexpr uint32_t WasmCallerTLSOffsetBeforeCall =
+    wasm::FrameWithTls::callerTLSOffset() + ShadowStackSpace;
+constexpr uint32_t WasmCalleeTLSOffsetBeforeCall =
+    wasm::FrameWithTls::calleeTLSOffset() + ShadowStackSpace;
+
 // The public entrypoint for emitting assembly. Note that a MacroAssembler can
 // use cx->lifoAlloc, so take care not to interleave masm use with other
 // lifoAlloc use if one will be destroyed before the other.